### ISSUE TITLE
Release read lock on a write action in EDT

### ISSUE THREAD

**Issue by fastasturtle** [johnthagen/clion-cppcheck#15]
Hi @johnthagen, thank you for the awesome plugin!

There is a performance issue that we observe occasionally in user reports: the annotation thread is waiting for the cppcheck process to finish under a read action. So when the write action is about to happen in the UI thread, it has to wait till the process finish, as the background thread is waiting on [`errorThread.join()`](https://github.com/johnthagen/clion-cppcheck/blob/master/src/com/github/johnthagen/cppcheck/CppcheckInspection.java#L184) and can't be canceled (i.e. by calling `ProgressManager#checkCanceled()`).

I'm attaching the problematic stack trace: [threadDump-20180329-183446.txt](https://github.com/johnthagen/clion-cppcheck/files/1863550/threadDump-20180329-183446.txt)
. You can reproduce this yourself by providing a never-ending executable instead of the cppcheck in the plugin options, forcing it to launch and performing any action that requires write access (i.e. "Reformat file").

The easy fix is to busy-wait for the external process, calling `checkCanceled()` periodically. There is some infrastructure in IntelliJ codebase to help with this, see e.g. https://github.com/JetBrains/intellij-community/blob/master/python/src/com/jetbrains/python/sdk/PySdkUtil.java#L153.

Please let me know if you need more information about it.

**Comment by johnthagen** [comment]
@fastasturtle Thanks for the detailed information. Unfortunately I'm mostly in a maintence mode in terms of what I can fix in my spare time for this plugin. I'd happily accept PRs (especially from Jetbrains). This issue looks to be out of scope for what I can handle on my own.
